python/packages/autogen-agentchat/src/autogen_agentchat/agents/_assistant_agent.py

```diff
@@ -215,12 +227,14 @@ def __init__(
         self,
         name: str,
         model_client: ChatCompletionClient,
+        model_context: ChatCompletionContext = BufferedChatCompletionContext(0),
```

Comment:
To match previous behavior you want an unbounded buffered context, this change has a buffer size of zero

We can either add the ability for the buffer to be unbounded or we can add a new impl that supports this. 

@ekzhu  thoughts?

Reply:
I think the default setting should be None. And for default we create a buffered context that is unbounded. But I am up for alternative ideas as well such as toke-limited context, using the context size information from model client.

Reply:
if I understand correctly, setting up the buffer_size to 0 will make the BufferedChatCompletionContext to be unbounded since the only time it's being used is in 
```
messages = self._messages[-self._buffer_size :]
```
is my understanding correct? or it's just better to create a different class for unbounded?

Reply:
Interesting, I think that's a bug in the model context since the documented contract is that that parameter is the buffer size. Ie number of messages contained in the buffer

Reply:
I created a different class for unbounded. Could you please have a look when you have time?